src/module_event_handlers.c

```diff
@@ -74,63 +76,69 @@ static bool _GraphContext_NameContainsTag(const GraphContext *gc) {
 
 // Calculate how many virtual keys are needed to represent the graph.
 static uint64_t _GraphContext_RequiredMetaKeys(const GraphContext *gc) {
-	uint64_t required_keys = 0;
 	RG_Config config = Config_GetModuleConfig();
-	required_keys += ceil((double)Graph_NodeCount(gc->g) / config.entities_threshold);
-	required_keys += ceil((double)Graph_EdgeCount(gc->g) / config.entities_threshold);
-	required_keys += ceil((double)Graph_DeletedNodeCount(gc->g) / config.entities_threshold);
-	required_keys += ceil((double)Graph_DeletedEdgeCount(gc->g) / config.entities_threshold);
-	return required_keys;
+	// If no limitation, return 0. The graph can be encoded in a single key.
+	if(config.entities_threshold == UNLIMITED) return 0;
+	uint64_t entities_count = Graph_NodeCount(gc->g) + Graph_EdgeCount(gc->g) + Graph_DeletedNodeCount(
+								  gc->g) + Graph_DeletedEdgeCount(gc->g);
+	if(entities_count == 0) return 0;
+	uint64_t key_count = ceil((double)entities_count / config.entities_threshold) - 1;
+	return key_count >= 0 ? key_count : 0;
 }
 
 static void _CreateGraphMetaKeys(RedisModuleCtx *ctx, GraphContext *gc) {
 	uint meta_key_count = _GraphContext_RequiredMetaKeys(gc);
 	bool graph_name_contains_tag = _GraphContext_NameContainsTag(gc);
 	for(uint i = 1; i <= meta_key_count; i++) {
+		char *uuid = UUID_New();
 		RedisModuleString *meta_rm_string;
 		/* Meta keys need to be in the exact shard/slot as the graph context key, to avoid graph sharding - we want to save all  the graph keys on the same shard.
```

Comment:
Extra space after: "we want to save all"
Also "to avoid graph sharding" at the target.